XamarinCommunityToolkit/Behaviors/ExceededLengthValidationBehavior.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xamarin.Forms;
+
+namespace XamarinCommunityToolkit.Behaviors
+{
+    public class ExceededLengthValidationBehavior : BaseBehavior<Entry>
+    {
+        public static readonly BindableProperty MaxLengthProperty = BindableProperty.
+            Create(nameof(MaxLength), typeof(int), typeof(ExceededLengthValidationBehavior));
+
+        public int MaxLength
+        {
+            get => (int)GetValue(MaxLengthProperty);
+            set => SetValue(MaxLengthProperty, value);
+        }
+
+        public static readonly BindableProperty TextColorProperty = BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(NumericValidationBehavior));
+
+        public Color TextColor
+        {
+            get => (Color)GetValue(TextColorProperty);
+            set => SetValue(TextColorProperty, value);
+        }
+
+        protected override void OnAttachedTo(Entry bindable)
+        {
+            bindable.TextChanged += OnEntryTextChanged;
+        }        
+
+        protected override void OnDetachingFrom(Entry bindable)
+        {
+            bindable.TextChanged -= OnEntryTextChanged;
+        }
+
+        private void OnEntryTextChanged(object sender, TextChangedEventArgs e)
+        {
+            bool isValid = (e.NewTextValue.Length < MaxLength);
+            ((Entry)sender).TextColor = isValid ? Color.Default : (TextColor != null ? TextColor : Color.Red);
```

Comment:
I wonder what happens if people might have a custom text color fitting their styling. Let's say you have a gray tone as text color. So if we add this behavior and always set it to `Color.Default` for them if valid it would override their own styling. Perhaps we should store the current text color in a private variable on attaching and set it to that when valid?

Reply:
I will review the logic again.